Sources/MessagingPush/Config/MessagingPushConfigBuilder.swift

```diff
@@ -0,0 +1,112 @@
+    /// Initializes new `MessagingPushConfigBuilder` with required configuration options.
+    /// - Parameters:
+    ///   - cdpApiKey: CDP API Key required for NotificationServiceExtension only to track metrics
+    public init(cdpApiKey: String) {
+        self.cdpApiKey = cdpApiKey
+    }
+
+    /// Enable automatic fetching of device token by the SDK without the need to write custom code by the customer.
+    /// On fetching the token, SDK will auto-register the device. This value is `true` by default.
+    @discardableResult
+    public func autoFetchDeviceToken(_ value: Bool) -> MessagingPushConfigBuilder {
+        autoFetchDeviceToken = value
+        return self
+    }
+
+    /// Automatic tracking of push events will automatically generate `opened` and `delivered` metrics for
+    /// push notifications sent by Customer.io. This value is `true` by default.
+    @discardableResult
+    public func autoTrackPushEvents(_ value: Bool) -> MessagingPushConfigBuilder {
+        autoTrackPushEvents = value
+        return self
+    }
+
+    /// Display push notifications sent by Customer.io while app is in foreground. This value is `true` by default.
+    @discardableResult
+    public func showPushAppInForeground(_ value: Bool) -> MessagingPushConfigBuilder {
+        showPushAppInForeground = value
+        return self
+    }
+
+    /// Builds and returns `MessagingPushConfigOptions` instance from the configured properties.
+    public func build() -> MessagingPushConfigOptions {
+        let configOptions = MessagingPushConfigOptions(
+            cdpApiKey: cdpApiKey,
+            autoFetchDeviceToken: autoFetchDeviceToken,
+            autoTrackPushEvents: autoTrackPushEvents,
+            showPushAppInForeground: showPushAppInForeground
+        )
+
+        return configOptions
+    }
+}
+
+public extension MessagingPushConfigBuilder {
+    /// Constants used to map each of the options in MessagingPushConfigOptions.
+    enum Keys: String {
+        case autoFetchDeviceToken
+        case autoTrackPushEvents
+        case showPushAppInForeground
+    }
+
+    /// Constructs `MessagingPushConfigOptions` by parsing and applying configurations from provided dictionary.
+    @available(iOSApplicationExtension, unavailable)
+    static func build(from dictionary: [String: Any]) -> MessagingPushConfigOptions {
+        let builder = MessagingPushConfigBuilder()
+
+        if let autoFetchDeviceToken = dictionary[Keys.autoFetchDeviceToken.rawValue] as? Bool {
+            builder.autoFetchDeviceToken(autoFetchDeviceToken)
+        }
+        if let autoTrackPushEvents = dictionary[Keys.autoTrackPushEvents.rawValue] as? Bool {
+            builder.autoTrackPushEvents(autoTrackPushEvents)
+        }
+        if let showPushAppInForeground = dictionary[Keys.showPushAppInForeground.rawValue] as? Bool {
+            builder.showPushAppInForeground(showPushAppInForeground)
+        }
+
+        return builder.build()
+    }
+}
```

Comment:
[Same comment](https://github.com/customerio/customerio-ios/pull/559#discussion_r1501143564)

Reply:
[Responded](https://github.com/customerio/customerio-ios/pull/559#discussion_r1502207125). We will be keeping it for now.